giddy/markov.py

```diff
@@ -615,6 +619,20 @@ def __init__(self, y, w, k=4, m=4, permutations=0, fixed=True,
                 y_int.append(list(map(label_dict.get, yi)))
             self.class_ids = np.array(y_int)
             self.lclass_ids = self.class_ids
+        elif manual_lag is not False: 
+            assert len(self.manual_lag) == len(y), "`lags must be the same length as the observation data"
+            merged = list(itertools.chain.from_iterable(y))
+            classes = np.unique(merged)
+            self.classes = classes
+            self.k = len(classes)
+            self.m = self.k
+            label_dict = dict(zip(classes, range(self.k)))
+            y_int = []
+            for yi in y:
+                y_int.append(list(map(label_dict.get, yi)))
+            self.class_ids = np.array(y_int)
+            self.lclass_ids = np.unique(self.manual_lag)
+            self.k = len(np.unique(self.manual_lag))
```

Comment:
a duplicate assignment of `self.k` (line 627)?

Reply:
hm.. I saw that... but this is similar to
https://github.com/pysal/giddy/blob/13fae6c18933614be78e91a6b5060693bea33a04/giddy/markov.py#L610